lib/types.nix

```diff
@@ -54,6 +54,11 @@ rec {
         right = lib.warn "maybeRaw.nestedTypes: `right` is a deprecated alias for `rawLua`." rawLua;
         inherit rawLua elemType;
       };
+      inherit (elemType)
+        getSubModules
+        getSubOptions
+        substSubModules
```

Comment:
I think a proper `substSubModules` is a little more complex than just re-using the elem-type's